src/main/java/hudson/tasks/JavadocArchiver.java

```diff
@@ -156,15 +156,35 @@ public String getIconFileName() {
          * Serves javadoc.
          */
         public void doDynamic(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
-            DirectoryBrowserSupport dbs = new DirectoryBrowserSupport(this, new FilePath(dir()), getTitle(), HELP_PNG, false);
-            if (new File(dir(), NOFRAMES_INDEX).exists() && Boolean.valueOf(
-                    System.getProperty(JavadocArchiver.class.getName() + ".useFramelessIndex", "true"))) {
-                /* If an overview-summary.html exists, serve that, unless the system property evaluates to false */
-                dbs.setIndexFileName(NOFRAMES_INDEX);
+            final File basedir = dir();
+            DirectoryBrowserSupport dbs = new DirectoryBrowserSupport(this, new FilePath(basedir), getTitle(), HELP_PNG, false);
```

Comment:
Looks like a no-op change?

Reply:
Yes, squashing the number of calls to make to `dir()`.